src/serializer/ResidualHeapVisitor.js

```diff
@@ -925,12 +927,20 @@ export class ResidualHeapVisitor {
         invariant(newValue);
         this.visitValue(newValue);
         residualBinding.modified = true;
-        // This should be enforced by checkThatFunctionsAreIndependent
-        invariant(
-          !residualBinding.additionalFunctionOverridesValue ||
-            residualBinding.additionalFunctionOverridesValue === functionValue,
-          "We should only have one additional function value modifying any given residual binding"
-        );
+        let otherFunc = residualBinding.additionalFunctionOverridesValue;
+        if (otherFunc !== undefined && otherFunc !== functionValue) {
+          let otherNameVal = otherFunc.$Get("name", otherFunc);
+          let otherNameStr = otherNameVal instanceof StringValue ? otherNameVal.value : "unknown function";
+          let funcNameVal = functionValue.$Get("name", functionValue);
+          let funNameStr = funcNameVal instanceof StringValue ? funcNameVal.value : "unknown function";
+          let error = new CompilerDiagnostic(
+            `Variable ${modifiedBinding.name} written to in optimized function ${funNameStr} conflicts with write in another optimized function ${otherNameStr}`,
+            funcNameVal.expressionLocation,
+            "PP1001",
```

Comment:
The error message does not match the wiki: https://github.com/facebook/prepack/wiki/PP1001

Reply:
The wiki has now been updated.